topdown/array.go

```diff
@@ -33,6 +35,43 @@ func builtinArrayConcat(a, b ast.Value) (ast.Value, error) {
 	return arrC, nil
 }
 
+func builtinArraySlice(a, i, j ast.Value) (ast.Value, error) {
+	arrA, err := builtins.ArrayOperand(a, 1)
+	if err != nil {
+		return nil, err
+	}
+
+	startIndex, err := builtins.IntOperand(i, 1)
+	if err != nil {
+		return nil, err
+	}
+
+	stopIndex, err := builtins.IntOperand(j, 2)
+	if err != nil {
+		return nil, err
+	}
+
+	// Don't allow negative indices for slicing
+	if startIndex < 0 || stopIndex < 0 {
+		return nil, fmt.Errorf("Invalid slicing operation: negative indices not allowed")
+	}
+
+	// stopIndex can't be less than startIndex
+	if stopIndex < startIndex {
+		return nil, fmt.Errorf("Invalid slicing operation: stopIndex can't be less than startIndex")
+	}
+
+	arrB := make(ast.Array, 0, stopIndex-startIndex)
```

Comment:
Since we're not deep copying here, we may as well juts slice the array:

```
arrb := arrA[startIndex:stopIndex]
```
Make sure the startindex and stopindex are clamped.